_dev/apps/ui/src/store/modules/product-feed/actions.ts

```diff
@@ -1,54 +1,81 @@
-import {FullState, RequestState} from '@/store/types';
-import appGetters from '@/store/modules/app/getters-types';
-import {ProductIssue} from '@/components/render-issues/types';
-import ProductsStatusType from '@/enums/product-feed/products-status-type';
 
 type Context = ActionContext<State, FullState>;
 
 // ToDo: Get DTO type from API sources
-export const createProductFeedApiPayload = (settings:any) => ({
-  autoImportTaxSettings: settings.autoImportTaxSettings,
-  shippingSetup: settings.shippingSetup,
-  targetCountries: settings.targetCountries,
-  ...(
-    (settings.shippingSetup === ShippingSetupOption.ESTIMATE) ? {
-      rate: settings.rate,
-      estimateCarriers: settings.estimateCarriers,
-    } : {}
-  ),
-  ...(
-    (settings.shippingSetup === ShippingSetupOption.IMPORT) ? {
+export const createProductFeedApiPayload = (settings: ProductFeedAPIPayload): ProductFeedAPIPayload => {
+  // const productFeedApiPayload: Partial<ProductFeedAPIPayload>= {
+  const productFeedApiPayload: ProductFeedAPIPayload = {
+    autoImportTaxSettings: settings.autoImportTaxSettings,
+    shippingSetup: settings.shippingSetup,
+    targetCountries: settings.targetCountries,
+
+    // biome-ignore lint/suspicious/noExplicitAny: TYPE:MUST_FIX
+    attributeMapping: formatMappingToApi(settings.attributeMapping as any),
+    selectedProductCategories: settings.selectedProductCategories,
+    requestSynchronizationNow: settings.requestSynchronizationNow,
+  };
+  // TODO: refactor so it is Typescript valid
+  const filterShippingSettings = (s: ShopShippingInterface) => (
+    (s.collection !== ShopShippingCollectionType.CARRIERS || (!!s.properties.active && !s.properties.deleted))
+      && (!s.properties.country_ids || settings.targetCountries?.some((tc: string) => s.properties.country_ids.includes(tc))));
+
+  switch (settings.shippingSetup) {
+    case ShippingSetupOption.ESTIMATE:
+      productFeedApiPayload.rate = settings.rate;
+      productFeedApiPayload.estimateCarriers = settings.estimateCarriers?.map(filterEstimateCarriers);
+      break;
+    case ShippingSetupOption.IMPORT:
       // Send in payload data related to active carriers and active countries on shop
-      shippingSettings: settings.shippingSettings?.filter((s) => (
-        (s.collection !== 'carriers' || (!!s.properties.active && !s.properties.deleted))
-        && (!s.properties.country_ids
-          || settings.targetCountries.some((tc: string) => s.properties.country_ids.includes(tc)))),
-      ),
-      additionalShippingSettings: settings.additionalShippingSettings,
-    } : {}
-  ),
-  attributeMapping: formatMappingToApi(settings.attributeMapping),
-  selectedProductCategories: settings.selectedProductCategories,
-  requestSynchronizationNow: settings.requestSynchronizationNow,
-});
+      productFeedApiPayload.shippingSettings = settings.shippingSettings?.filter(filterShippingSettings);
```

Comment:
If we want to improve the types used in the product feed configuration, we could make this attribute mandatory when the Import is chosen.

Reply:
same as for `estimateCarriers`.
The `shippingSettings` is set only if `settings.shippingSetup === ShippingSetupOption.IMPORT`
As for the previous one, if that's ok, I'll make the change